assets/src/gaussholder.js

```diff
@@ -144,26 +144,22 @@ export default function () {
 	 * Render all placeholders on the page
 	 */
 	return function () {
-		let images = document.getElementsByTagName( 'img' );
-
-		for ( let i = images.length - 1; i >= 0; i-- ) {
-			let img = images[i];
-
-			// Ensure the blank GIF has loaded first
-			if ( img.complete ) {
-				handleElement( img );
-			} else {
-				img.onload = function () {
-					handleElement( this );
-				};
-			}
-		}
+		const images = document.getElementsByTagName( 'img' );
+
+		const options = {
+			rootMargin: '1200px'
+		};
 
-		loadLazily = images;
-		scrollHandler();
+		const imagesObserver = new IntersectionObserver( (entries) => {
+			const visibleImages = entries
+				.filter( ( { isIntersecting } ) => isIntersecting === true );
 
-		if ( loadLazily.length > 0 ) {
-			window.addEventListener( 'scroll', scrollHandler );
-		}
+			visibleImages.forEach( ( { target } ) => loadOriginal( target ) );
+		}, options);
+
+		Array.from( images ).forEach( ( img ) => {
```

Comment:
`Array.from` (and `IntersectionObserver`) aren't available in IE. What kind of fallback should we provide so that IE users don't just lose all the images on the site?

Reply:
I've separated the code into a few more modules. The main file in src/gaussholder.js will choose a handler based on the Intersection API availability now.